lib/asm-parser.js

```diff
@@ -271,7 +271,7 @@ export class AsmParser extends AsmRegex {
         const commentOnly = /^\s*(((#|@|\/\/).*)|(\/\*.*\*\/)|(;\s*)|(;[^;].*)|(;;.*\S.*))$/;
 
         const commentOnlyNvcc = /^\s*(((#|;|\/\/).*)|(\/\*.*\*\/))$/;
-        const sourceTag = /^\s*\.loc\s+(\d+)\s+(\d+).*/;
+        const sourceTag = /^\s*\.loc\s+(\d+)\s+(\d+)\s+(.*)/;
```

Comment:
Should this not specifically be `(\d*)` instead of `(.*)`?

Reply:
In some outputs the column is absent, but other characters (symbol name?) are present. I want the match to still succeed with the proper file name and line set. For example: https://github.com/compiler-explorer/compiler-explorer/blob/main/test/filters-cases/icc.hellow.asm#L16

Reply:
I'll add a test case to demonstrate this.

Reply:
Sounds good! thanks